### ISSUE TITLE
Project documentation

### ISSUE THREAD

**Issue by mpryor** [notree-md/notree#42]
If notree is going to be providing FOSS libraries for developers to build their own markdown/notes applications, we will want to provide really solid documentation.

We'll want to consider the different *types* of documentation as well.

See: https://documentation.divio.com/ or https://www.writethedocs.org/videos/eu/2017/the-four-kinds-of-documentation-and-why-you-need-to-understand-what-they-are-daniele-procida/

1. Information-oriented documentation: I'm thinking API docs and auto-generated internal class/function/etc docs. (a lot of these types of docs could probably be automated)
2. Goal-oriented documentation: Some how-to/getting started guides. We could host these types of docs using the SaaS notree product itself? 

There are other types we might consider but having these would be a good start

**Comment by jollyjerr** [comment]
I agree information-oriented and goal-oriented documentation will be most important for us at this point! Let's host our docs on our sass product for sure - that will be a good use case. Perhaps the docs can live in the open source repo as a directory of markdown files and then the sass can use the github provider to grab them?
